.github/workflows/tests.yml

```diff
@@ -144,3 +150,129 @@ jobs:
+          NEOFS_HTTP_GATE: ${{ vars.NEOFS_HTTP_GATE }}
+          STORE_OBJECTS_CID: ${{ vars.STORE_OBJECTS_CID }}
+          LIFETIME: ${{ vars.LIFETIME }}
+          PATH_TO_FILES_DIR: ${{ env.REPL_DATA_DIR }}
+          REPLACE_OBJECTS: True
+          REPLACE_CONTAINER_CONTENTS: False
+
+      - name: Run tests
+        env:
+          OUTPUT_CONTAINER_URL: ${{ steps.gh_push_to_neofs_with_replace_objects.outputs.OUTPUT_CONTAINER_URL }}
+          REPORT_DIR: ${{ env.REPL_DATA_DIR }}
+        run: |
+          source venv/bin/activate && pytest test_downloads.py --base_url="$OUTPUT_CONTAINER_URL" --report_dir="$REPORT_DIR" --data_dir_prefix=../$REPORT_DIR
+        working-directory: ./tests
+
+      - name: Prepare directory name for tests with entire container rewrite
+        shell: bash
+        env:
+          TIMESTAMP: ${{ steps.date.outputs.timestamp }}
+        run: |
+          echo "REWRITE_CONT_DIR=${{ github.run_number }}-$TIMESTAMP-$(uuidgen)-cont-rewrite" >> $GITHUB_ENV
+
+      - name: Create a directory tree for tests with objects replacement
+        shell: bash
+        run: |
+          mkdir "$REWRITE_CONT_DIR"
+
+      - name: Move files to the directory tree for tests with url prefix
+        shell: bash
+        env:
+          SOURCE_DIR: ${{ env.TESTS_DATA_DIR }}
+          DEST_DIR: ${{ env.REWRITE_CONT_DIR }}
+        run: |
+          rsync -av "$SOURCE_DIR" "$DEST_DIR"
+
+      - name: Run gh-push-to-neofs with pre objects replacement
+        id: gh_push_to_neofs_with_pre_cont_rewrite
+        uses: ./
+        with:
+          NEOFS_WALLET: ${{ secrets.NEOFS_WALLET }}
+          NEOFS_WALLET_PASSWORD: ${{ secrets.NEOFS_WALLET_PASSWORD }}
+          NEOFS_NETWORK_DOMAIN: ${{ vars.NEOFS_NETWORK_DOMAIN }}
+          NEOFS_HTTP_GATE: ${{ vars.NEOFS_HTTP_GATE }}
+          STORE_OBJECTS_CID: ${{ vars.STORE_OBJECTS_CID }}
+          LIFETIME: ${{ vars.LIFETIME }}
+          PATH_TO_FILES_DIR: ${{ env.REWRITE_CONT_DIR }}
+          REPLACE_OBJECTS: False
+          REPLACE_CONTAINER_CONTENTS: False
+
+      - name: Modify test data
+        shell: bash
+        env:
+          DATA_DIR: ${{ env.REWRITE_CONT_DIR }}
+        run: |
+          find $DATA_DIR -type f -name '*.txt' -exec sed -i '$ s/$/_cont_rewrite/' {} +
+
+      - name: Run gh-push-to-neofs with objects replacement
+        id: gh_push_to_neofs_with_cont_rewrite
+        uses: ./
+        with:
+          NEOFS_WALLET: ${{ secrets.NEOFS_WALLET }}
+          NEOFS_WALLET_PASSWORD: ${{ secrets.NEOFS_WALLET_PASSWORD }}
+          NEOFS_NETWORK_DOMAIN: ${{ vars.NEOFS_NETWORK_DOMAIN }}
+          NEOFS_HTTP_GATE: ${{ vars.NEOFS_HTTP_GATE }}
+          STORE_OBJECTS_CID: ${{ vars.STORE_OBJECTS_CID }}
+          LIFETIME: ${{ vars.LIFETIME }}
+          PATH_TO_FILES_DIR: ${{ env.REWRITE_CONT_DIR }}
+          REPLACE_OBJECTS: True
+          REPLACE_CONTAINER_CONTENTS: True
+
+      - name: Run tests
```

Comment:
Do we check the contents afterwards? We need to ensure that only the new contents is present and all the old one is deleted.

Reply:
I added a test to check the number of objects in the container and the number of previously uploaded files.